recipe/thirdparty/providers/apple.go

```diff
@@ -178,9 +178,11 @@ func verifyAndGetClaimsAppleIdToken(idToken string, clientId string) (jwt.MapCla
 	jwksURL := "https://appleid.apple.com/auth/keys"
 
 	// Create the keyfunc options. Refresh the JWKS every hour and log errors.
-	refreshInterval := time.Hour
 	options := keyfunc.Options{
-		RefreshInterval: refreshInterval,
+		// https://github.com/supertokens/supertokens-golang/issues/155
+		// This causes a leak as the pointer to JWKS would be held in the goroutine and
+		// also results in compounding refresh requests
+		// RefreshInterval: time.Hour,
 	}
```

Comment:
Please see how other people do this and if this needs to be done in a "global" scope instead of doing it here.

Reply:
Updated to use a cache, as that's more efficient compared to this. Have used the standard golang's approach for global stuff.